enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,343 @@
+- Enable Openshift users to request a pod be put in a user namespace
+- Update SCC to take user namespaces into account when choosing the security profile of a container
+- Add support for users to run podman within an Openshift pod without being privileged.
+
+### Non-Goals
+
+- Anything not related to user namespaces or running nested containers.
+
+## Proposal
+
+There are two pieces to this proposal:
+- Extend SCC to be aware of the hostUsers field:
+    - Add a new field HostUsers to SCC, which will be Allowed by default
+    - Add a new SCC to the default list: restricted-v3
+        - This SCC will be identical to restricted-v2, but have HostUsers set to Disallowed
+    - Add a new SCC to the default list: nonroot-v3
+        - This SCC will be identical to nonroot-v2, but have HostUsers set to Disallowed
+    - Add a new SCC to the default list: nested-container. It will have:
+        - SELinux context set to MustRunAs.Type: container_engine_t
+        - RunAsUserStrategy: RunAsAny
+        - RequiredDropCapabilities: None
+        - HostUsers: Disallowed
+        - And otherwise mirror the restricted-v2 profile
+- Add the features UserNamespacesPodSecurityStandards and ProcMountType to the list that qualify a cluster as TechPreviewNoUpgrade
+    - UserNamespacesSupport will be on by default, as the feature is beta in 1.30.
+
+### Workflow Description
+
+- (TechPreview) A cluster admin turns on the `UserNamespacesSupport`, `UserNamespacesPodSecurityStandards` and `ProcMountType` features with the featuregate CRD.
+- The cluster reconciles, enabling these features in the kubelet and apiservers.
+- Pod authors can update their pods and set the `hostUsers` field to `false`.
+    - Depending on the SCC the namespace is allowed to use, the pod author may be required to do so.
+- Pod authors can also set the `procMount` field to `Unmasked` (if they also set `hostUsers` to `false`).
+- Pod authors could then run nested containers, or other operations that can benefit from a user namespace and unmasked `/proc`
+
+### API Extensions
+
+- Add HostUsers to SCC. This relies on approval from the apiserver team.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### SecurityContextConstraint Updates
+
+##### `HostUsers` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // hostUsers determines if the policy allows host users in containers.
+       // Valid values are "Allowed", "Disallowed" and omitted.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default,
+       // which is subject to change over time.
+       // The current default is "Allowed".
+       // +openshift:enable:FeatureGate=UserNamespacesPodSecurityPolicy
```

Comment:
This line doesn't actually match any of the upstream gates, are you intending to gate this field on one of the upstream gates? Is it useful in a scenario where all are not enabled?

Reply:
whoopsies, I meant `UserNamespacesPodSecurityStandards`

Reply:
If you require to do so, we can tie it to an ANY or ALL semantic for the gates, but you'll need to represent the upstream gates within the openshift API repository to do so (I think what you're describing in this doc requires that anyway)

Reply:
> you'll need to represent the upstream gates within the openshift API repository to do so

is this what https://github.com/openshift/api/pull/1940 does or is there more to it?

Reply:
Yep, I think you've nailed that